src/libraries/Common/tests/System/Security/Cryptography/CngPlatformProviderKey.cs

```diff
@@ -14,13 +14,17 @@ internal sealed class CngPlatformProviderKey : IDisposable
             string keySuffix = null,
             [CallerMemberName] string testName = null,
             params CngProperty[] additionalParameters)
+            : this(algorithm, CreateDefaultParameters(), keySuffix, testName, additionalParameters)
         {
-            CngKeyCreationParameters cngCreationParameters = new CngKeyCreationParameters
-            {
-                Provider = CngProvider.MicrosoftPlatformCryptoProvider,
-                KeyCreationOptions = CngKeyCreationOptions.OverwriteExistingKey,
-            };
+        }
 
+        public CngPlatformProviderKey(
+            CngAlgorithm algorithm,
+            CngKeyCreationParameters cngCreationParameters,
+            string keySuffix = null,
+            [CallerMemberName] string testName = null,
+            params CngProperty[] additionalParameters)
```

Comment:
I don't like this constructor, because if `additionalParameters` are provided then it modifies `cngCreationParameters`.

Reply:
Based on usages later, I expect you're using this class because the dispose deletes the key.

To me, that means this class should just be renamed to `SelfDeletingCngKey` or `CngSelfDeletingKey`, or even `CngKeyWrapper`.  If need be for preview 6 deadline reasons, this can be done as a followup.  But we shouldn't have a name "PlatformProvider" for a thing that we're extending to work with other providers.

This ctor should be removed, and replaced with a static method for using the software KSP.  E.g.  `public static SelfDeletingCngKey CreateSoftwareKey`, which takes whatever options you need.  That'll probably require making a private ctor that takes the prebuilt CngKeyCreationParameters; leave the existing one alone.

In an ideal world (but perhaps not a time constrained one), the existing public ctor would also be removed and replaced with a `CreatePlatformKey` (or some suitable name) with parameters that look like the current ctor.